### DISCUSSION TITLE
is it possible to add extension?

### DISCUSSION THREAD

**Discussion by ChunduriS** [arkypita/LaserGRBL#1437] [General]
Currently, I am using Inkscape to generate G-Code from image for Laser-GRBL.

Every time it is difficult for me to convert image into G-Code and transfer to the GRBL.

So, i need to generate the G-code with in GRBL.  Is it possible to add Inkscape as extension to the Laser-GRBL.

processes i need to do:-
Step 1:- Input image 
Step 2:- identifying the points
Step 3:- generate a G-Code by using points 
Step 3:- G-Code to laser microcontroller for engraving or Cut operation
Step 4:- output.

**Comment by gmmanonymus111** [comment]
Why don't you use `Vectorize` option in LaserGRBL? It can create paths, and you don't have to mess in Inkscape. Or, LaserGRBL can even engrave images directly.

**Reply by ChunduriS** [reply]
Hi Alex,

Can you tell me how to use this Vectorize option in GRBL.

**Reply by gmmanonymus111** [reply]
Once you've opened LaserGRBL, click on `Open File`:
![image](https://user-images.githubusercontent.com/9658243/116529295-17795b00-a8dd-11eb-9817-4f15e83b1f34.png)
Select the image you want to engrave, then click `Open` (Image can be `PNG`, `JPG`, `JPEG` etc, but make sure it is not an `SVG` vector file)
After opening an image, you can choose the processing method. It can be `1bit BW dithering`, `Vectorize`, `Centerline`, or if you don't want LaserGRBL to make any processing on your image `Passthrough`. Now we are going to use `Vectorize`. If you image, like mine has too much white space, you can auto-crop it (marked with a red arrow):
![image](https://user-images.githubusercontent.com/9658243/116532431-86a47e80-a8e0-11eb-86f3-c14749ae2719.png)
The finished preview (`Vectorize` and cropped):
![image](https://user-images.githubusercontent.com/9658243/116532634-beabc180-a8e0-11eb-85a1-e57d46d6e307.png)
Click `Next`, then set your engraving speed and power:
![image](https://user-images.githubusercontent.com/9658243/116532986-2bbf5700-a8e1-11eb-909f-b0c8aa481ea5.png)
Then click `Create`, the vectorized image should appear in LaserGRBL's engraving preview:
![image](https://user-images.githubusercontent.com/9658243/116533179-5a3d3200-a8e1-11eb-9c71-e0d8ce86639e.png)
More info about LaserGRBL usage can be [found here](https://lasergrbl.com/usage/)
